### PR TITLE
tdx: Add artefacts from the latest TDX tools release into main

### PR DESCRIPTION
**Pull request by fidencio**
**NOTE: This changes will break the TDX CI on the CCv0 branch, as those are NOT backward compatible**

This series adds to the `main` branch the artefacts released as part of https://github.com/intel/tdx-tools/releases/tag/2023ww01.

The series is extensive, must be careful reviewed, and is the very first step to merging the `CCv0` content into the `main` branch.

Please, take a look at each commit / commit message as those may include important details / descriptions to help the reviewer.

### CODE REVIEW

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:268]: Nit: `export qemu_tarball_name="kata-static-qemu-${qemu_suffix}.tar.gz"`

and on line 269: `"qemu-${qemu_suffix}"`

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:184]: ah, `$(yq r $versions_yaml FOO` can be replaced by `get_from_kata_deps FOO`. The later function is use throughout the kata-deploy-binaries.sh script

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:185]: Just checking that it will use the same `kata_config_version` for all the kernel flavors, otherwise this `kernel_kata_config_version` should be parameterized as well.

File: tools/packaging/static-build/cache_components_main.sh
Comment (wainersm) [inline tools/packaging/static-build/cache_components_main.sh:15]: Typo: kernel-arm-experimetnal but it is not your fault :)

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:391]: Shouldn't be the tarball name here?

File: src/runtime/virtcontainers/hypervisor_linux_amd64.go
Comment (wainersm) [inline src/runtime/virtcontainers/hypervisor_linux_amd64.go:22]: Maybe you should treat the error better. The directory might exist but you got no permission to stat and returning that tdx guest protection is false is misleading.

File: tools/packaging/static-build/cache_components_main.sh
Comment (fidencio) [reply tools/packaging/static-build/cache_components_main.sh:15]: I introduced this, so it's my fault in the end. :-)

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:184]: Done!

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:185]: If we bump a config for any kernel we need to bump this file, so this is the same for all flavours.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:391]: It should, nice catch, thanks!

File: src/runtime/virtcontainers/hypervisor_linux_amd64.go
Comment (fidencio) [reply src/runtime/virtcontainers/hypervisor_linux_amd64.go:22]: As `containerd-shim-v2` runs as `root`, I sincerely think we're fine on how we're dealing with this.

File: src/runtime/pkg/govmm/qemu/qemu.go
Comment (fidencio) [reply src/runtime/pkg/govmm/qemu/qemu.go:351]: It seems there are a few different ways to load a firmware, and each one depends on the firmware, on the machine type, and a few other things.

For QEMU TDX we need to explicitly set the bios, and I agree that it wouldn't require the check on `Valid()`, but I'm not confident to remove it and end up breaking different (and more common) use-cases.

**Comment by fidencio** [comment]
I'd like to have the TDX CI up and running before merging this one.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (arronwy) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:241]: `$versions_yaml` is not required since it will get by itself.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (arronwy) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:381]: default name  `edk2.tar.cz` => `edk2.tar.gz`?

File: src/runtime/Makefile
Comment (arronwy) [inline src/runtime/Makefile:128]: duplicate empty line

**Review by arronwy — APPROVED** [review]
Thanks @fidencio LGTM! Verified on TDX Server with 2023WW01 host kernel.

**Comment by fidencio** [comment]
I'm removing the `do-not-merge` label as a TDX machine has been added as a self-hosted runner.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:448]: Your very first commit on this PR added extra space before ";;" :)

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:388]: Another possibility for the the job's name is `component_name` rather than `ovmf-${ovmf_type}`? What would be more flexible for future changes?

Comment (wainersm) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:388]: Never mind. It would make the code more complex because we would need to convert ovmf_type -> component_name in other scripts like the one for caching.

Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:388]: This one I'm not that much in favour, @wainersm.

What we'll end up having is:
* ovmf-tdx
* ovmf-sev-snp
* ovmf-x86_64 (the default one, which can also be used for `sev-es`)

I think as it's right now is very much flexible for future changes, without any other work apart from copying what's done for the tdx function.

File: src/runtime/Makefile
Comment (wainersm) [inline src/runtime/Makefile:125]: hmm.... the merge from CCv0 to main has a potential to turn this Makefile more convoluted than it is today. We should start thinking on how to break it into smaller makefile snippets to be included in.

File: src/runtime/config/configuration-qemu-tdx.toml.in
Comment (fidencio) [reply src/runtime/config/configuration-qemu-tdx.toml.in:43]: That's a fair point, I will do.

File: src/runtime/Makefile
Comment (fidencio) [reply src/runtime/Makefile:125]: I have been going back and forth on this, mainly on a way to have a different snippet that could only be loaded specifically for a runtime-class, but I sincerely don't see how it can happen unless we:
* symlink the default configuration
* create snippets under that new name

The reason for that is because once a snippet is in, it'll be loaded, and right now I do believe users will still want to have the "vanilla" snippets and the TEEs ones. :-/

File: .github/workflows/run-k8s-tests-on-tdx.yaml
Comment (wainersm) [inline .github/workflows/run-k8s-tests-on-tdx.yaml:22]: Do you think in protect that machine by allowing jobs only with a certain label like "ok-to-test"?

Comment (fidencio) [reply .github/workflows/run-k8s-tests-on-tdx.yaml:22]: I've been going back and forth on this, and the PRs will only automatically run for members of the Kata Containers organisation, which are the same ones that can add an "ok-to-test" label.

With this in mind, I kinda prefer to trust that no malicious attempt will come from inside the Kata Containers org.

**Review by wainersm — APPROVED** [review]
I left a couple of comments/suggestion, none that blocks this PR. Ready to go!

Hey, thanks once again @fidencio . I appreciated that you made the scripts more flexible for future variations of QEMU, etc, to be built!

**Pull request merged**
